src/Core/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -150,41 +150,92 @@ public Type GetColumnSystemType(string columnName)
+        private void LinkingTableProcessing(
             string targetEntityName,
             string relatedSourceAlias,
             BaseSqlQueryStructure subQuery)
         {
             SourceDefinition sourceDefinition = GetUnderlyingSourceDefinition();
             DatabaseObject relatedEntityDbObject = MetadataProvider.EntityToDatabaseObject[targetEntityName];
             SourceDefinition relatedEntitySourceDefinition = MetadataProvider.GetSourceDefinition(targetEntityName);
+
             if (// Search for the foreign key information either in the source or target entity.
-                sourceDefinition.SourceEntityRelationshipMap.TryGetValue(
-                    EntityName,
-                    out RelationshipMetadata? relationshipMetadata)
-                && relationshipMetadata.TargetEntityToFkDefinitionMap.TryGetValue(
-                    targetEntityName,
-                    out List<ForeignKeyDefinition>? foreignKeyDefinitions)
-                || relatedEntitySourceDefinition.SourceEntityRelationshipMap.TryGetValue(
-                    targetEntityName, out relationshipMetadata)
-                && relationshipMetadata.TargetEntityToFkDefinitionMap.TryGetValue(
-                    EntityName,
-                    out foreignKeyDefinitions))
+                 sourceDefinition.SourceEntityRelationshipMap.TryGetValue(
+                     EntityName,
+                     out RelationshipMetadata? relationshipMetadata)
+                 && relationshipMetadata.TargetEntityToFkDefinitionMap.TryGetValue(
+                     targetEntityName,
+                     out List<ForeignKeyDefinition>? foreignKeyDefinitions)
+                 || relatedEntitySourceDefinition.SourceEntityRelationshipMap.TryGetValue(
+                     targetEntityName, out relationshipMetadata)
+                 && relationshipMetadata.TargetEntityToFkDefinitionMap.TryGetValue(
+                     EntityName,
+                     out foreignKeyDefinitions))
             {
                 Dictionary<DatabaseObject, string> associativeTableAndAliases = new();
                 // For One-One and One-Many, not all fk definitions would be valid
                 // but at least 1 will be.
-                // Identify the side of the relationship first, then check if its valid
+                // Identify the side of the relationship first, then check if it's valid
                 // by ensuring the referencing and referenced column count > 0
                 // before adding the predicates.
                 foreach (ForeignKeyDefinition foreignKeyDefinition in foreignKeyDefinitions)
                 {
                     // First identify which side of the relationship, this fk definition
                     // is looking at.
-                    if (foreignKeyDefinition.Pair.ReferencingDbTable.Equals(DatabaseObject))
+                    if (EntityName == targetEntityName)
```

Comment:
Why is self-joining case considered under `LinkingTableProcessing()`?

Reply:
good catch, not intended. removed.